apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/PipelineApplication.java

```diff
@@ -102,11 +102,9 @@ public static void main(String[] args) throws InterruptedException {
 
     appMetricsReporter.start(1, TimeUnit.HOURS);
 
-    HikariDataSource dataSource =
-        DatabaseUtils.createDataSource(
-            appConfig.getDatabaseOptions(),
-            appMetrics,
-            2 * appConfig.getCcwRifLoadOptions().getLoadOptions().getLoaderThreads());
+    // Create the application state.
+    PipelineApplicationState appState =
+        new PipelineApplicationState(appMetrics, appConfig.getDatabaseOptions());
```

Comment:
Nice.  This is a big improvement even without thinking about test performance since it allows all of the jobs to use the same connection pool with a configurable size.